libsel4/include/sel4/bootinfo.h

```diff
@@ -10,6 +10,23 @@
 #include <sel4/bootinfo_types.h>
 #include <sel4/macros.h>
 
+/* The constant CONFIG_MAX_NUM_BOOTINFO_UNTYPED_CAPS has been removed from
+ * CMake, it automatically adapts to the boot info frame's size now. To keep
+ * compatibility, the define is provide here now. Getting the boot info frame
+ * size within the kernel is easy, as BI_FRAME_SIZE_BITS exists. For userland,
+ * this is currently not exposed and there seems an assumption that this is
+ * always one page. We define BI_FRAME_SIZE_BITS for userland here.
```

Comment:
Isn't it better to just expose the kernel's BI_FRAME_SIZE_BITS?

Reply:
Yes. I was wondering if this should be made a config option. Or if we just adapt the boot info frame size dynamically to the required size. But then we might have to communicate the size to the root task also, which requires a new fiel somewhere. However, it's also implicitly available due to the number of untyped slots and implies size of the descriptor array. 
Which reminds me that BI_FRAME_SIZE_BITS is fixed to 4KiB anyway because of the extra boot info that is defined to start 4 KiB after the boot info. I have to add this in the comment. And unless we break compatibility, there is a hard limit at 500 untyped descriptors in the 4 KiB page.

Reply:
Actually, what about having a new `CONFIG_BI_FRAME_SIZE_BITS` instead that defaults to 12? And then have a 
```
SEL4_COMPILE_ASSERT(seL4_BootInfo_fits, CONFIG_BI_FRAME_SIZE_BITS >= seL4_PageBits)
#define BI_FRAME_SIZE       LIBSEL4_BIT(BI_FRAME_SIZE_BITS)
```
in the header. Rhe documentation would say that `CONFIG_BI_FRAME_SIZE_BITS` can be increased if `untypedList[]` is too small. 
That may break some useland code that has the hard-coded assumption the boot info frame is 4 KiB, but this needs to be adapted then. Unfortunately, there is no way to communicate this value to userland dynamically. But even for a pre-compiled kernel the headers provided with it make `CONFIG_BI_FRAME_SIZE_BITS` (and thus `BI_FRAME_SIZE`) available for the root task.

Reply:
The boot info frame has to be one page, and changing this is not trivial. So no chance to make this trivially configurable for now. 240 untyped desciptors (500 on 32 bit) is really a hard limit, but it seems this is not hit anywhere besides some experimental configurations. As a follow-up change, we could allocate the boot info frame and the extra boot info together as a bunch of pages without special alignment needs. That would then allow having the the boot info take more than one page and the kernel boot memory allocation would get a bit simpler also. Seem there is no reason why extra boot info needs an alignment greater then one page either.